### PR TITLE
Track addon/file status changes better

### PR DESCRIPTION
**Pull request by kumar303**

### CODE REVIEW

File: apps/addons/models.py [hunk 1]
```diff
@@ -2341,4 +2341,14 @@ def track_status_change(old_attr={}, new_attr={}, **kw):
     new_status = new_attr.get('status')
```
Comment (mstriemer) [inline apps/addons/models.py]: Doesn't this function get an `instance` in `**kw`? I think that would be better than passing a `dict` to `track_addon_status_change` here and using `instance.__dict__` in the other call site.

File: apps/addons/tests/test_models.py [hunk 2]
```diff
@@ -2390,29 +2391,57 @@ def create_addon(self, **kwargs):
...
+        with patch('addons.models.track_addon_status_change') as mock_:
             addon.update(status=amo.STATUS_PUBLIC)
-        mock_incr.assert_called_with(
-            'addon_status_change.status_{}'.format(amo.STATUS_PUBLIC)
-        )
+
+        d = addon.__dict__.copy()
+        d['status'] = amo.STATUS_PUBLIC
+        assert (sorted(mock_.call_args[0][0].keys()) ==
+                sorted(k for k in d.keys() if not k.startswith('__')))
```
Comment (mstriemer) [inline apps/addons/tests/test_models.py]: Feels like these tests are fighting the fact that a `dict` is being passed in. Even though it isn't great if you can't use an instance I'd suggest passing two keyword arguments to it instead.

**Comment by mstriemer** [comment]
I'd rather see the instance used in those functions but it looks good otherwise.

r+wc and passing build

**Comment by kumar303** [comment]
oh! I think the signal does pass an instance, I'll try and use that.

File: apps/addons/models.py (hunk 1 above)
Comment (kumar303) [inline apps/addons/models.py]: it does indeed. Fixed! Thanks. Dealing with the dicts was giving me a headache.

**Pull request merged**